tests/test_globaliostream.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+    tests/test_globaliostream.cpp -- Usage of global scoped_output_redirect
+
+    Copyright (c) 2017 Henry F. Schreiner
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+
+#include <pybind11/iostream.h>
+#include "pybind11_tests.h"
+#include <iostream>
+
+TEST_SUBMODULE(globaliostream, m) {
+    // test_evals
+    m.def("redirect_output", []() {
+            return py::capsule(new py::scoped_output_redirect(
```

Comment:
Redesigned, and made available in the iostream.h header. I've tried to tweak the names a bit too to be clearer.